### PR TITLE
ENG-3182: Add Hybrid extractor

### PR DESCRIPTION
**Pull request by tvinagre**

### CODE REVIEW

File: tycho-indexer/src/extractor/revert_buffer.rs [hunk 1]
```diff
@@ -271,18 +271,22 @@ where
 {
     /// Looks up buffered protocol state updates for the provided keys. Returns a map of updates and
     /// a list of keys for which updates were not found in the buffered blocks.
+    ///
+    /// Clippy thinks it is a complex type that is difficult to read
```
Comment (zizou0x) [inline tycho-indexer/src/extractor/revert_buffer.rs]: `///` makes this comment appear in the function description no?

File: tycho-indexer/src/extractor/evm/hybrid.rs
Comment (louise-poole) [inline tycho-indexer/src/extractor/evm/hybrid.rs:1]: It looks like we're creating a whole new extractor for this. Is there a point then to keeping the old extractors around?  They're now outdated no?

Comment (tvinagre) [reply tycho-indexer/src/extractor/evm/hybrid.rs:1]: I'll remove them - but wanted to do it in a separate PR, after everything is up and running

Comment (louise-poole) [reply tycho-indexer/src/extractor/evm/hybrid.rs:1]: Then I think when we do that we remove the word 'hybrid' from the names here. It's weird to have stuff called 'hybrid' where there are no 'non-hybrid' ones in comparison.

Comment (tvinagre) [reply tycho-indexer/src/extractor/evm/hybrid.rs:1]: Doesn't bother me that much bc it's a hybrid - can process both VM and Native protocols, but I see what you mean - we can change!

**Pull request merged**